features/proposed/FxA-70-kpi-dashboards/README.md

```diff
@@ -0,0 +1,414 @@
+and so that detailed event data
+can be deleted after a resonable retention period.
+
+### Proposed Queries
+
+Below are some proposed strawman queries
+for how we might calculate the KPIs
+on a daily basis.
+We'll obviously need to experiment
+with the details of these
+on live data.
+
+Active Daily Users:
+
+    SELECT APPROXIMATE COUNT (DISTINCT uid)
+    FROM events
+    WHERE timestamp > [previous UTC midnight]
+      AND timestamp <= [current UTC midnight];
+
+Active Monthly Users:
+
+    SELECT APPROXIMATE COUNT (DISTINCT uid)
+    FROM events
+    WHERE timestamp > [UTC midnight 30 days ago]
+      AND timestamp <= [current UTC midnight];
+
+Active Daily Multi-Device Users:
+
+    SELECT APPROXIMATE COUNT (DISTINCT eNow.uid)
+    FROM events AS eNow
+      INNER JOIN events AS ePrev
+      ON eNow.uid = ePrev.uid
+    WHERE eNow.timestamp > [previous UTC midnight]
+      AND eNow.timestamp <= [current UTC midnight]
+      AND ePrev.timestamp > [UTC midnight 5 days ago]
+      AND ePrev.timestamp < eNow.timestamp
+      AND eNow.deviceId IS NOT NULL
+      AND ePrev.deviceId IS NOT NULL
+      AND eNow.deviceId != ePrev.deviceID;
+
+Active Monthly Multi-Device Users:
+
+    SELECT APPROXIMATE COUNT (DISTINCT eNow.uid)
+    FROM events AS eNow
+      INNER JOIN events AS ePrev
+      ON eNow.uid = ePrev.uid
+    WHERE eNow.timestamp > [UTC midnight 30 days ago]
+      AND eNow.timestamp <= [current UTC midnight]
+      AND ePrev.timestamp > [UTC midnight 5 days before eNow.timestamp]
+      AND ePrev.timestamp < eNow.timestamp
+      AND eNow.deviceId IS NOT NULL
+      AND ePrev.deviceId IS NOT NULL
+      AND eNow.deviceId != ePrev.deviceID;
+
+(In practice, we'll probably want to synthesize
+a concrete table of
+"userids that counted as multi-device on each day",
+to avoid the expensive joins above.)
+
+Daily time to device connection percentiles:
+
+    SELECT MIN(median)
+    FROM (
+      SELECT seqTime,
+        PERCENTILE_CONT([0.25|0.5|0.75])
+          WITHIN GROUP (ORDER BY seqTime) OVER() AS median
+      FROM events
+      WHERE type = "device.connected"
+        AND timestamp > [previous UTC midnight]
+        AND timestamp <= [current UTC midnight]
+        AND service = "sync"
+        AND newUser = [0|1]
+    );
+
+Daily percent completion
+of device connection:
+
+    SELECT
+      100.0 * completed / begun as perc\_completed,
+      (SELECT COUNT(\*) 
+       FROM events
+       WHERE type = 'sequence.begin'
+         AND timestamp > [previous UTC midnight]
+         AND timestamp <= [current UTC midnight]
+         AND service = "sync"
+      ) AS begun,
+      (SELECT COUNT(\*)
+       FROM events
+       WHERE type = 'device.connected'
+         AND timestamp > [previous UTC midnight]
+         AND timestamp <= [current UTC midnight]
+         AND service = "sync"
+      ) AS completed;
+
+
+## Work Breakdown
+
+We'll need to:
+
+* [ ] Emit new "sequence.begin" activity event from fxa-content-server.
```

Comment:
I'm going to keep it in the spec because we'll need to seek approval from data stewards for this extra trying-together of events, but will call out a phased implementation approach to begin with.